lib/katello/plugin.rb

```diff
@@ -211,7 +211,7 @@
     :content_facet_attributes => [:content_view_id, :lifecycle_environment_id, :content_source_id,
                                   :host, :kickstart_repository_id],
     :subscription_facet_attributes => [:release_version, :autoheal, :purpose_usage, :purpose_role, :service_level, :host,
-                                       {:installed_products => [:product_id, :product_name, :arch, :version]}, :facts, :hypervisor_guest_uuids => [], :purpose_addons => []]
+                                       {:installed_products => [:product_id, :product_name, :arch, :version]}, :facts, :hypervisor_guest_uuids => [], :purpose_addon_ids => []]
```

Comment:
Whenever I update purpose addons from the UI I see this in the logs:

```
2019-06-18T01:03:01 [D|app|ffe23364] Unpermitted parameter: :purpose_addons
```

It would be nice if we could omit this. I think we need to add back purpose_addons here, and I think we need purpose_addon_ids too.

Reply:
Adding the :purpose_addons => [] back does not seem to help because it takes more precedence over purpose_addon_ids, so an instance of a String is being accepted rather than Katello::PurposeAddon and this breaks the functionality.

Reply:
As you suggested, deleting the :purpose_addons key after updating the :purpose_addon_ids in the before action under hosts_controller_extensions.rb  work.